pkg/executor/nginx/nginx_executor.go

```diff
@@ -0,0 +1,120 @@
+package nginx
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+//Executor :
+type Executor interface {
+	PrepareForNginxRun(nginxConfigPath string) *exec.Cmd
+	HandleExit(exitCode int, pid int) int
+	StartLogRotate(pid int, timeInMs int64)
+}
+
+type nginxExecutor struct {
+	nginxExitHandler
+	nginxLogRotate
+}
+
+type nginxExitHandler struct {
+}
+
+type nginxLogRotate struct {
+	paths           []string
+	rotateAfterSize int64
+}
+
+//NewNginxExecutor :
+func NewNginxExecutor(rotateAfterSize int64, logfiles []string) Executor {
+	return nginxExecutor{
+		nginxExitHandler{},
+		nginxLogRotate{
+			paths:           logfiles,
+			rotateAfterSize: rotateAfterSize,
+		},
+	}
+}
+
+func (m nginxExecutor) PrepareForNginxRun(nginxConfigPath string) *exec.Cmd {
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("exec nginx -g 'daemon off;' -c %s", nginxConfigPath))
```

Comment:
Hvorfor bruker du sh som wrapper her? Eneste nytten er vel parsing av kommandolinjeparametre. Forskjellen på exec og ikke er heller ikke helt triviell, så jeg tror det er bedre å bare droppe sh for forståelse for de som kommer etter.

Reply:
På grunn av parsingen. Skal se om jeg får gjort det på en annen måte

Reply:
Har testet litt igjen. Klarer ikke å kjøre den uten en shell kontekt på grunn av 'daemon off' direktivet jeg sender inn

Reply:
Det skal være mulig, men du escaper feil tenker jeg.

Reply:
Det gikk uten ''